bouncerscript/utils.py

```diff
@@ -103,9 +91,43 @@
 
 
 async def api_show_location(context, product_name):
+    """Function to query the API for specific locations of a product"""
+    data = {}
+    return await api_call(context, "location_show?product=%s" %
+                          quote(product_name), data=data)
+
+
+async def api_update_alias(context, alias, product_name):
+    """Function to update an aliases to a specific product"""
+    data = {
+        "alias": alias,
+        "related_product": product_name,
+    }
+
+    return await api_call(context, "create_update_alias", data)
+
+
+async def does_product_exist(context, product_name):
+    """Function to check if a specific product exists in bouncer already by
+    parsing the XML returned by the API endpoint."""
+    res = await api_show_product(context, product_name)
+
+    try:
+        xml = parseString(res)
+        # bouncer API returns <products/> if the product doesn't exist
+        products_found = len(xml.getElementsByTagName("product"))
+        log.info("Products found: {}".format(products_found))
+        return bool(products_found)
+    except Exception as e:
```

Comment:
Nit: Can we pinpoint a specific set of exceptions? Otherwise, errors like `IndexError` (which can be caused by a bad `str.format()` call in `log.info` will wrongly induce a non-existing product.

Reply:
Good point, will follow-up with a fix catching more exceptions.